DataStructures/Graph/IDirectedWeightedGraph.cs

```diff
@@ -14,6 +14,8 @@ public interface IDirectedWeightedGraph<T>
 
         bool AreAdjacent(Vertex<T> startVertex, Vertex<T> endVertex);
 
+        double AdjacentDistance(Vertex<T> startVertex, Vertex<T> endVertex);
```

Comment:
Is it necessary? It seems like it's unnecessary

Reply:
I need to to be able to get the distances between vertices, since the adjacency matrix is private I thought this would be ideal.